src/Adapter/Product/CommandHandler/SetProductSuppliersHandler.php

```diff
@@ -89,65 +88,41 @@ public function __construct(
     /**
      * {@inheritdoc}
      */
-    public function handle(UpdateProductSuppliersCommand $command): array
+    public function handle(SetProductSuppliersCommand $command): array
     {
-        $this->assertCommandIsNotEmpty($command);
         $productId = $command->getProductId();
+        $defaultSupplierIdValue = $command->getDefaultSupplierId()->getValue();
+        $this->assertDefaultSupplierIsOneOfProvidedSuppliers($command);
 
-        if (null !== $command->getProductSuppliers()) {
-            $this->updateProductSuppliers($productId, $command->getProductSuppliers());
-        }
-
-        $deletedAllProductSuppliers = is_array($command->getProductSuppliers()) && empty($command->getProductSuppliers());
-
-        if (!$deletedAllProductSuppliers) {
-            $this->handleDefaultSupplierUpdate($command);
-        }
+        $this->setProductSuppliers($productId, $command->getProductSuppliers());
+        $this->updateProductDefaultSupplier($productId, $defaultSupplierIdValue);
```

Comment:
Watch out, when you update supplier_id and supplier_reference you also need to update the `wholesale_price` in the product table :)

Reply:
Can you explain a little more ?
In what case i need to reset the wholesale_price?
I remember you said that it must be reset when product has combinations.
Or its simple like that - i reset it to 0 always when updating default supplier?

Reply:
When you select a default supplier without having a combination, it takes the supplier reference and the wholesale price.

More information here https://github.com/PrestaShop/PrestaShop/pull/20198#issuecomment-665150376